lib/shape.mli

```diff
@@ -1,8 +1,8 @@
 type point = { x : float; y : float }
-type circle = { c : point; radius : float }
-type ellipse = { c : point; rx : float; ry : float }
-type polygon = point list
-type line = { a : point; b : point }
+type circle = { c : point; radius : float; color : Color.color }
+type ellipse = { c : point; rx : float; ry : float; color : Color.color }
+type polygon = { vertices : point list; color : Color.color }
+type line = { a : point; b : point; color : Color.color }
```

Comment:
Ideally we need not expose the internals of shape types in the mli file. But, I see that they're used in the `Transform` module, so we can't do away with it. Makes me wonder whether we should put them all in the Shape module.

Reply:
Put all the transforms in shape.ml? I don't think that would bulk it up too much so I wouldn't have a problem with it, but I think I do have a small preference for breaking up projects like this.